settings.go

```diff
@@ -23,6 +23,11 @@ type Settings struct {
 	DisableSort bool
 	// DisableJoin ignore the "join" query if true.
 	DisableJoin bool
+	// DisableSearch ignore the "search" query if true.
+	DisableSearch bool
+
+	// FieldsSearch allows search for these fields
+	FieldsSearch []string
```

Comment:
For this feature, I would see two approaches:
- Let the client define the fields used for the search (and prevent searching on fields that are in the blacklist)
- Hardcode the fields used for the search on the server-side (what you are doing in this case)

The second option requires a bit more setup on the server-side and wouldn't work at all without providing custom settings. I think the first option offers more flexibility.